src/components/activity-header/nav-pages.tsx

```diff
@@ -1,46 +1,91 @@
 import React from "react";
 import IconHome from "../../assets/svg-icons/icon-home.svg";
-import ReactPaginate from "react-paginate";
+import { Page } from "../../types";
 
 import "./nav-pages.scss";
-import { Page } from "../../types";
+
+const kMaxPages = 11;
 
 interface IProps {
   pages: Page[];
   currentPage: number;
   onPageChange: (page: number) => void;
+  lockForwardNav?: boolean;
 }
 
 export class NavPages extends React.PureComponent <IProps> {
   render() {
     return (
-      <div className="nav-pages" data-cy="nav-pages" >
+      <div className="nav-pages" data-cy="nav-pages">
         {this.renderHomePageButton()}
-        <ReactPaginate
-          previousLabel={"<"}
-          nextLabel={">"}
-          breakLabel={""}
-          pageCount={this.props.pages.length}
-          onPageChange={this.handlePaginate}
-          marginPagesDisplayed={0}
-          pageRangeDisplayed={10}
-          containerClassName={"paginate-container"}
-          pageLinkClassName={"page-button"}
-          activeClassName={"current"}
-          activeLinkClassName={"current"}
-          forcePage={this.props.currentPage - 1}
-          previousClassName={`${this.props.currentPage === 0 ? "disabled" : ""}`}
-          previousLinkClassName={"page-button"}
-          nextLinkClassName={"page-button"}
-        />
+        {this.renderPreviousButton()}
+        {this.renderButtons()}
+        {this.renderNextButton()}
       </div>
     );
   }
 
+  private renderPreviousButton = () => {
+    const { currentPage } = this.props;
+    return (
+      <button
+        className={`page-button ${currentPage === 0 ? "disabled" : ""}`}
+        onClick={this.handleChangePage(currentPage - 1)}
+      >
+        {"<"}
+      </button>
+    );
+  }
+  private renderNextButton = () => {
+    const { currentPage, pages, lockForwardNav } = this.props;
+    const totalPages = pages.length;
+    return (
+      <button
+        className={`page-button ${currentPage === totalPages || lockForwardNav ? "disabled" : ""}`}
+        onClick={this.handleChangePage(currentPage + 1)}
+      >
+        {">"}
+      </button>
+    );
+  }
+
+  private renderButtons = () => {
+    const { currentPage, pages, lockForwardNav } = this.props;
+    const totalPages = pages.length;
+    const maxPagesLeftOfCurrent = currentPage - 1;
+    const maxPagesRightOfCurrent = totalPages - currentPage;
+    let minPage = 1;
+    let maxPage = totalPages;
+    if (maxPagesLeftOfCurrent < 5) {
+      maxPage = Math.min(totalPages, currentPage + 5 + (5 - maxPagesLeftOfCurrent));
+    } else if (maxPagesRightOfCurrent < 5) {
+      minPage = Math.max(1, currentPage - 5 - (5 - maxPagesRightOfCurrent));
+    } else if (totalPages > kMaxPages) {
+      minPage = currentPage - 5;
+      maxPage = currentPage + 5;
```

Comment:
Seems like all of these `5`s could be a constant like `kMaxButtonsPerSide`, which is probably computable as Math.floor(`kMaxNavigationButtons / 2)`. Then if we ever wanted to change the max number of navigation buttons everything should just work.

Reply:
Good call.  Fixed.